lib/kubernetes-deploy/kubernetes_resource/custom_resource.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+require 'jsonpath'
+module KubernetesDeploy
+  class CustomResource < KubernetesResource
+    TIMEOUT_MESSAGE_DIFFERENT_GENERATIONS = <<~MSG
+      This resource's status could not be used to determine rollout success because it was still out of date
+      (.metadata.generation != .status.observedGeneration) after #{timeout}s.
```

Comment:
It's referencing the `timeout` _class_ method on `KubernetesResource` :( 
From what I see `debug_message` will already show the timeout for a given resource so we can just drop the reference here but it will still be visible in the logs

Reply:
For ref:
<img width="1221" alt="screen shot 2019-01-18 at 1 10 36 pm" src="https://user-images.githubusercontent.com/31742287/51404792-78246580-1b22-11e9-97dc-6569e9f5cfd3.png">

Reply:
maybe "it was still out of date when the timeout was reached", or rephrase in the present tense? I think you're totally right that we don't need to repeat the actual timeout, but the sentence sounds a little odd to me with the end removed.

Reply:
Also `Final status: <blank>` needs to be fixed there